### ISSUE TITLE
design change in communication

### ISSUE THREAD

**Issue by N1c084** [stuartpittaway/diyBMSv4#127]
hi ,
 what do you think about these mod to be able to have a better reliability and capacity in comm ?
VSS is the voltage from the other module, requested by the new opte 6N137

Module v440:
![image](https://user-images.githubusercontent.com/101097267/199770919-27270a6a-a837-4bc1-87df-8ad67f51df66.png)
Contrôler:
![image](https://user-images.githubusercontent.com/101097267/199771280-79f7e4ef-d55c-43ec-b68c-4f58b305c6ea.png)

**Comment by stuartpittaway** [comment]
Hi.  Are you planning to swap to a 3 pin connector for the comms ?

**Comment by N1c084** [comment]
Hi yes it is , I will made pcb with 14s and the wire will be between contrôler and 14s pcb > Hi.  Are you planning to swap to a 3 pin connector for the comms ?

**Comment by stuartpittaway** [comment]
How about using something like VOH1016AD ?

https://www.mouser.co.uk/ProductDetail/Vishay-Semiconductors/VOH1016AD?qs=PqoDHHvF648sjq9xbZMnUg%3D%3D

That is almost a drop in replacement for the existing chip.  Wouldn't need to change connector then either.

**Comment by stuartpittaway** [comment]
What's the problem you are trying to solve with the higher comms speed?

**Comment by N1c084** [comment]
Needs to drive 154 modules in one system
So I want to be the more efficient and secure with com.
This optocoupler support high speed compare to the actual

Le dim. 6 nov. 2022 à 16:17, Stuart Pittaway ***@***.***> a
écrit :

**Comment by stuartpittaway** [comment]
Ok.  Are you really going to be working with +500VDC on this?

**Comment by N1c084** [comment]
This is the same , need 3 wires at output side.

Le dim. 6 nov. 2022 à 16:15, Stuart Pittaway ***@***.***> a
écrit :

**Comment by N1c084** [comment]
No 😅 11p 14s 48V 😀

Le dim. 6 nov. 2022 à 16:30, Stuart Pittaway ***@***.***> a
écrit :

**Comment by stuartpittaway** [comment]
That's a relief!!

Why do you need 3 wires?

A module is just driving the anode and cathode of the chip in the next module along the chain.

![image](https://user-images.githubusercontent.com/1201909/200179785-d5c3b13c-b666-4001-914d-04015cccf8b6.png)

**Comment by stuartpittaway** [comment]
or a chip like the Toshiba TLP2301 - this should be good for up to 1Mbps (although the ATTINY won't go that fast!)

**Comment by N1c084** [comment]
Because all voltage need to come from the other board the stay isolated
from the board , ground /vcc / output from second board

Le dim. 6 nov. 2022 à 16:33, Stuart Pittaway ***@***.***> a
écrit :

**Comment by stuartpittaway** [comment]
The isolator is on the RX so can be on the modules, so only two wires needed, just like now.

**Comment by N1c084** [comment]
> or a chip like the Toshiba TLP2301 - this should be good for up to 1Mbps (although the ATTINY won't go that fast!)

I will try with this chip, thanks
There is no new hardware release that coming in the  next 6 month? I just received a batch of attiny 841

**Comment by stuartpittaway** [comment]
Yes, I'm going to release a new module version using a different attiny chip, because the 841 is difficult to order.

**Comment by N1c084** [comment]
there is no functional change or new features?

**Comment by stuartpittaway** [comment]
No function changes, the new attiny has some hardware improvement
